### PR TITLE
First Release of "Taste Jamaica"

### PR DESCRIPTION
**Pull request by ahangarha**
## Description

In this project, we made a website to show a list of Jamaican meals that users can like and leave comments for. The core features are:

- Get list of meals from API
- Get detail of a chosen meal and show on popup
- Like individual meals and update the like status on Involvement API
- Add new comments for each meal and store on Involvement API
- Write unit test for counting items and comments

Please view this short video we made about our project [here](https://drive.google.com/file/d/1qaYGdQVeB3aOAglSsiZQibMMapr34kRm/view?usp=sharing)

### General requirements

- [x] There are no linter errors.
- [x] Followed the **Github flow**.
- [x] The documentation is professional.

### HTML/CSS requirements

- [x] Followed best practices for HTML & CSS
- [x] Followed best practices for JavaScript

### Project requirements

https://github.com/microverseinc/curriculum-javascript/blob/main/group-capstone/js_capstone.md

---

Thank you for taking the time to review this pull request.

### CODE REVIEW

File: src/modules/api-utils.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+export function getData(url) {
+  return fetch(url).then((res) => res.json());
+}
+
+export function sendData(url, data) {
+  return fetch(url, {
+    method: 'POST',
+    headers: {
```
Comment (elmar8287) [inline src/modules/api-utils.js]: - [x] Please, consider using ES6 syntax while creating the functions. I have checked all `.js` files, so I did not see arrow functions used. I know, you can easily fix it, so go-go guys! 👍  Please, fix it in all `.js` files 💯

File: src/modules/core-functions.js [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+    const COMMENT_API = INV_API_BASE + INV_API_KEY + COMMENT_ENDPOINT + parameter;
+    getData(COMMENT_API).then((commentsFromAPI) => {
+      if (commentsFromAPI.error) commentsFromAPI = [];
+      foodList.addComments(id, commentsFromAPI);
+      resolve();
+    });
+  });
+}
+
+function postComment(id, input, textarea) {
```
Comment (elmar8287) [inline src/modules/core-functions.js]: - [x] Please, consider using an arrow function, aka ES6 syntax 🥇

File: src/modules/core-functions.js [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+      commentWrapper.innerHTML += `<li class="comment">
+        <h4 class="comment-author">${data.username}</h4>
+        <p class="comment-message">${data.comment}</p>
+        </li> `;
+    }
+  });
+}
+
+export function displayPopUp(id) {
+  commentPopup.classList.add('show');
```
Comment (elmar8287) [inline src/modules/core-functions.js]: - [x] Please, consider using an arrow function, aka ES6 syntax 🥇

File: src/modules/core-functions.js [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+        </li> `;
+      });
+    } else {
+      commentWrapper.innerHTML = 'no comments';
+    }
+  });
+}
+
+export function likeFood(id) {
+  const url = INV_API_BASE + INV_API_KEY + LIKES_ENDPOINT;
```
Comment (elmar8287) [inline src/modules/core-functions.js]: - [x] Please, consider using an arrow function, aka ES6 syntax 🥇

File: src/modules/core-functions.js [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+      foodList.setLikes(id, newLikes);
+      const likeWrapper = document.getElementById(id);
+      const counterElement = likeWrapper.querySelector('.likes-counter');
+      counterElement.innerHTML = foodList.getLikesText(id);
+    }
+  });
+}
+
+export function showAllFood() {
+  // clear loading text
```
Comment (elmar8287) [inline src/modules/core-functions.js]: - [x] Please, consider using an arrow function, aka ES6 syntax 🥇

File: src/modules/core-functions.js [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+      likesFromAPI.forEach((likeObject) => {
+        foodList.setLikes(likeObject.item_id, likeObject.likes);
+      });
+      resolve();
+    });
+  });
+}
+
+export function displayitemCounter() {
+  const size = foodList.getItemsCount();
```
Comment (elmar8287) [inline src/modules/core-functions.js]: - [x] Please, consider using an arrow function, aka ES6 syntax 🥇

File: src/modules/core-functions.js [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+          name="comment"
+          rows="5"
+          placeholder="Your Comment"
+          required
+          id="messageField"
+        ></textarea>
+        <button id="button">Submit</button>
+      </form>
+    </div>`;
+
```
Comment (elmar8287) [inline src/modules/core-functions.js]: - [ ] Please, consider also fixing catching the empty comment adding. You can add a warning message maybe, if the inputs are empty and submit button will not actually submit.
 > Here I have entered an empty comment and I am able to do it:

**Review by elmar8287 — CHANGES_REQUESTED** [review]
Hi, team!

WOW, very nice project done, guys! I liked it very much. Regarding the video - that is a perfect presentation of the project. As you know, it should be a maximum of 5 min., but you have an additional 37 sec. - but really I enjoyed watching it, so the video is passed and approved, guys 👍  

Also, I want to highlight some points:

- No linter issues
- Nice descriptive PR 👍 
- Considering Gitflow! 💯 

### **Let us check other points!** 🥇 

Good job so far!

There are some issues that you still need to work on to go to the next project but you are almost there!

# Required Changes ♻️

- [ ] Check the comments under the review.

## Optional suggestions

- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/api-utils.js (hunk 1 above)
Comment (ahangarha) [reply src/modules/api-utils.js]: done

File: src/modules/core-functions.js (hunk 2 above)
Comment (ahangarha) [reply src/modules/core-functions.js]: done

File: src/modules/core-functions.js (hunk 4 above)
Comment (ahangarha) [reply src/modules/core-functions.js]: done

File: src/modules/core-functions.js (hunk 7 above)
Comment (ahangarha) [reply src/modules/core-functions.js]: Thanks for noticing this issue. it is fixed now.

**Review by Kingstalux — APPROVED** [review]
Hi @ahangarha and @daudi13 , 
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:
68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f78543078657a514755357843444a754350652f67697068792e676966

Good job implementing the required changes 👍 Your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :ship: 


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**